access_monitoring_rules: Add timezone to schedules spec

bernardjkim opened the pull request:
Supports: https://github.com/gravitational/teleport/issues/51682, https://github.com/gravitational/teleport/issues/55075
RFD: https://github.com/gravitational/teleport/pull/57334

This extends the access_monitoring_rules schedules spec with timezones.

This was previously omitted from https://github.com/gravitational/teleport/pull/59007 after choosing to store all schedules in the backend in UTC. However, after attempting to implement the timezone conversions on the frontend, I've realized this approach is not compatible with the frontend design for schedules.

There are a couple reasons why converting to UTC in the backend is not compatible with the design, but the main reason is that the frontend design limits 1 shift per day. This is a problem because valid schedules configured using the standard editor can easily be converted into a schedule that is not valid within the schedule editor.

For example, if a user configures the following schedule using the standard editor:
<img width="364" height="256" alt="Screenshot 2025-10-08 at 2 30 04 PM" src="https://github.com/user-attachments/assets/4cf35a94-e6ce-4991-9ea4-667c07a7db41" />

This schedule would then be converted into UTC in the backend, and stored as:
```yaml
schedules:
  default:
    time:
      shifts:
      - weekday: Monday
        start: 19:00
        end: 23:59
      - weekday: Tuesday
        start: 00:00
        end: 6:59
      - weekday: Tuesday
        start: 19:00
        end: 23:59
      - weekday: Wednesday
        start: 00:00
        end: 6:59
```

If the user attempts to edit this schedule again within the WebUI, they would not be able to use the standard editor because there are two shifts for Tuesday.

bernardjkim committed:
Add timezone to AMR schedules

marcoandredinis reviewed (commented):
Honestly, I don't like the idea of storing timezones in the backend.

First, there's the timezone naming: our WebUI can help here but what about other clients?
tsh, terraform, kube operator and other teleport-api clients
Should we use the abbreviation format or `continent/city` format?
(eg, WET vs Europe/Lisbon)
Given the printscreen, we use the former, but the unit tests use the latter.
Can we use both?

Then, there's the format we support: only abbreviated like WEST, PDT, and so on?
Or also the `continent/city` format?
If using abbreviated, can we set WET and WEST?
WET only applies during half of the year, and WEST during the other half.
What happens if a shift is set to WEST timezone but we are not in that part of the year (thus, using WET).
(the same applies to PDT/PST)

Should we only accept the format where we have a specific location (continent/city like format)?

Can you please add a couple more tests to exercise these edge cases?


The fact that the UI does not allow for shifts that overlap two sequential days should not deter us from providing a great UX to the end user.

As an example, we could use something like this to have an overall visualization of the configured shifts:

<img width="608" height="455" alt="Image" src="https://github.com/user-attachments/assets/a9a321b2-e0c3-4166-91b8-e227d9480e6b" />

bernardjkim committed:
Update documentation with accepted timezone values

bernardjkim committed:
Address feedback

- Allow empty timezone value
- Reference IANA in error message
- Add additional timeonze test cases

bernardjkim committed:
Fix test case

ptgott reviewed (approved):
Just a reminder that if we're going to release this in a minor version after backporting, we should specify in the docs which minor version the new fields require.
